File: src/main/java/io/pageobject/generator/locator/protractor/NameLocatorSource.java
package io.pageobject.generator.locator.protractor;

import io.pageobject.generator.GeneratorContext;
import io.pageobject.generator.attribute.HtmlAttributes;
import io.pageobject.generator.locator.AbstractLocatorSource;
import org.jsoup.nodes.Element;

import static io.pageobject.generator.attribute.Attributes.extractAttributeValueAllowingIndexExpression;

public class NameLocatorSource extends AbstractLocatorSource {

    @Override
    public String extractLocatorValue(Element element, GeneratorContext context) {
        return extractAttributeValueAllowingIndexExpression(element, HtmlAttributes.NAME);
    }

    public String locatorTemplate() {
        return "by.name('%s')";
    }

}

File: lib/locator/protractor/name-locator-source.js
"use strict";

const AbstractLocatorSource = require('../abstract-locator-source');
const _ = require('lodash');
const extractAttributeValueAllowingIndexExpression = require('../../attribute/attributes').extractAttributeValueAllowingIndexExpression;
const HtmlAttribute = require('../../attribute/html-attribute');

const template = _.template("by.name('<%= value %>')");

class NameLocatorSource extends AbstractLocatorSource {

  extractLocatorValue(context) {
    return extractAttributeValueAllowingIndexExpression(context.element, HtmlAttribute.NAME);
  }

  locatorTemplate() {
    return template;
  }

}

module.exports = NameLocatorSource;